staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -182,6 +186,7 @@ func NewNamedReflector(name string, lw ListerWatcher, expectedType interface{},
 		resyncPeriod:           resyncPeriod,
 		clock:                  realClock,
 		watchErrorHandler:      WatchErrorHandler(DefaultWatchErrorHandler),
+		stopHandle:             stopHandle,
```

Comment:
What happens if `stopHandle` is nil?

Reply:
This is a good point. Take a look at the most recent commit. The least intrusive solution I could think of was just to add a couple more constructors and creating a new stopHandle if it's nil. 

This way, the stopHandle should never be nil if the reflector is created via constructor. A couple concerns here are:
1. It feels a little chaotic to have 4 different constructors for the reflector. Open to ways of doing this cleaner. We could do a builder pattern here or use something like an Options struct, but that seems like a big change and not sure if it's worth doing here.
2. Similar to the comment in controller.go, if you're not using a constructor and just using the struct, there's still a chance you're gonna have a nil stopHandle. We could nil-check and default it in the `RunWithStopOptions()` function like I've commented, but it feels a little messy.